tools/c7n_azure/c7n_azure/filters.py

```diff
@@ -212,6 +205,16 @@ def get_metric_data(self, resource):
         self._write_metric_to_resource(resource, metrics_data, m)
 
         return m
+    
+    def get_metric_data(self, resource):
```

Comment:
a little confusing what is the difference between `self.filter` and `self.get_filter()`.. 
Maybe create a subclass for all Child resource metrics and use `self.parent_filter` instead of `self.get_filter() ?

Reply:
I thought about breaking out `get_filter()` but the override in the child class needs to modify the resource id field that is getting passed in and the filter field. This way we maintain one function that needs to be overridden instead of two.

@logachev I think the confusion might be that in this context `self.filter` is an Azure SDK concept and has nothing to do with the c7n concept of filter. An example for Cosmos DB databases would be `self.filter = "DatabaseName eq 'testdatabase'"` which only returns metrics for the given database

Reply:
Still feels weird to me to do this thing on each child as we'll have lots of places to update it if the API or our functionality changes:

```
class CosmosDBCollectionMetricFilter(MetricFilter):
    def get_metric_data(self, resource):
        if self.filter is None:
            parent_filter = "CollectionName eq '%s'" % resource['id']
        else:
            parent_filter = "%s and CollectionName eq '%s'" % (self.filter, resource['id'])
        return self.client.metrics.list(
            resource['c7n:parent-id'],
            timespan=self.timespan,
            interval=self.interval,
            metricnames=self.metric,
            aggregation=self.aggregation,
            filter=parent_filter
        )
```

Seems like worst case you could do this instead on each resource with less changes to the base and subclass

```
class CosmosDBCollectionMetricFilter(MetricFilter):
    def __init__(self, data, manager=None):
        super(CosmosDBCollectionMetricFilter, self).__init__(data, manager)
        self.filter = ' and '.join(("CollectionName eq '%s'" % resource['id'], self.filter))
        self.id_override =  resource['c7n:parent-id']
```

Then in the base you only have to change `resource['id']` to `self.id_override or resource['id']`

Or you could put the string modification in the base and just set the one field (all children have the `resource['c7n:parent-id']` so you can build the two fields you need if you set this:

```
class CosmosDBCollectionMetricFilter(MetricFilter):
    def __init__(self, data, manager=None):
        super(CosmosDBCollectionMetricFilter, self).__init__(data, manager)
        self.parent_filter_dimension = 'CollectionName'
```